### DISCUSSION TITLE
How to disable code-server-session when using subpath?

### DISCUSSION THREAD

**Discussion by zhaozhiming** [coder/code-server#6151] [Q&A] [answered]
Hi all, 
I start an inner API server in code-server, something like localhost:8888.
![image](https://user-images.githubusercontent.com/1436694/232705294-de9f523a-cf47-49bf-8a3f-d70338bfdcf1.png)

I can access success the API through `subpath` like  localhost:8080/proxy/8888 in the browser.
![image](https://user-images.githubusercontent.com/1436694/232705409-bca62e69-5fe6-465a-91ea-dfa93d039ec8.png)

But I can not access the API through curl in the terminal. I encounter the error: 'Found. Redirecting to ./../login?to=% '. 
![image](https://user-images.githubusercontent.com/1436694/232705514-6d09ba02-a3c1-4012-9860-931a0c6c3841.png)

I think the reason may be the lack of the code-server-session cookie when using the curl. When I add the cookie it works.
![image](https://user-images.githubusercontent.com/1436694/232705718-dbc4d9d4-a177-43fa-84cc-d829bac72ac4.png)

But I hope I can directly access the API without code-server-session using the curl in the terminal. 
How can I solve this problem? 
Thanks.

**Comment by code-asher** [comment]
If you are running the command in code-server's terminal there is no need to 
go through the proxy, right?  You could just directly `curl localhost:8888`.

**Reply by zhaozhiming** [reply]
Yes. I can directly `curl localhost:8888` in the code-server's terminal. 
But I want to call its internal services from outside the code-server. 
In my use case, I have 2 code-server for 2 different people. One is used to develop the frontend project. The other is used to develop the backend project. And the frontend will invoke the backend API to debug something.

**Reply by code-asher** [reply, marked as answer]
That makes sense.  Right now there is no way to bypass the auth check unfortunately.  Eventually we want to add the ability to make a URL public.

For now I think you would need to use something else to proxy the services or remove password authentication (possibly replacing it with some kind of external authentication).

**Reply by zhaozhiming** [reply]
Thanks. I will try your way.

**Comment by LYUYork** [comment]
anything else? how to make it work like these ways:

1.  localhost:8888?cookie=xxxx
2. localhost:8888?password=xxx. ps. the $password is the same in ~/.config/code-server/config.yaml

**Reply by code-asher** [reply]
This is not possible at the moment, but PRs for this are welcome: https://github.com/coder/code-server/issues/3546

**Comment by sarowarhosen03** [comment]
just use auth=none at confige.yml
